lib/ansible/plugins/connection/ssh.py

```diff
@@ -279,7 +278,25 @@
           cli:
             - name: private_key_file
               option: '--private-key'
-
+      private_key:
+          description:
+            - Private key contents in PEM format. Requires the C(SSH_AGENT) configuration to be enabled.
+          type: string
+          env:
+            - name: ANSIBLE_PRIVATE_KEY
+          vars:
+            - name: ansible_private_key
+          version_added: '2.19'
+      private_key_passphrase:
+          description:
```

Comment:
would it make sense to use the connection password instead? it would be overloading, but allows for prompting already, for those that don't want to serialize the passphrase nor use vault.

it is also how paramiko already works(ed?).

Reply:
Personally I am not a fan of overloading variables but I can see it being useful.

Should we keep `private_key_passphrase` and fall back to the connection password if passphrase is not given?

Reply:
I wouldn't, I prefer being more explicit.  The fact that paramiko did it was an implementation detail of paramiko and not our connection plugin.

Reply:
> an implementation detail of paramiko

It looks like it is a documented feature of both [paramiko](https://docs.paramiko.org/en/latest/api/client.html#paramiko.client.SSHClient.connect) and our [connection plugin](https://docs.ansible.com/ansible/devel/collections/ansible/builtin/paramiko_ssh_connection.html#parameter-password) so I thought that was intentional.

> I prefer being more explicit

+1

Moreover, the functionality would be limited like with the connection password in that only one key passphrase could be used for all keys?

Reply:
no, you would still have ansible_password, just as settable as this config, but sivel's point stands